### ISSUE TITLE
error when creating the weather widget

### ISSUE THREAD

**Issue by Nin3s** [ucf-senior-design/senior-design#242]
“Application error: a client-side exception has occurred (see the browser console for more information).”

this also broke my trip so it will not load for me anymore as of rn :[

**Comment by aamare1001** [comment]
![image](https://user-images.githubusercontent.com/26231761/231654091-ab282b84-9f3c-4089-84f7-ae38dd5613a1.png)
Just added weather to a new trip and currently working. will keep trying things to see when it stops

**Comment by Nin3s** [comment]
> ![image](https://user-images.githubusercontent.com/26231761/231654091-ab282b84-9f3c-4089-84f7-ae38dd5613a1.png)
> 
> Just added weather to a new trip and currently working. will keep trying things to see when it stops

wait didn't it originally show the current forecast + the next 5 days 🤔

**Comment by aamare1001** [comment]
Not sure, but i just did a trip for Topeka Kansas and the weather widget doesn't work, i wonder if there's a patterns with the locations

**Comment by aamare1001** [comment]
<img width="327" alt="image" src="https://user-images.githubusercontent.com/26231761/232122888-f8bf0b7d-2bae-4bc4-8bd8-644d6a620e5a.png">
Currently getting a 401 unauthorized error when accessing weather on deployment
https://openweathermap.org/faq#error401
<img width="658" alt="image" src="https://user-images.githubusercontent.com/26231761/232123143-f47f90d8-0d75-47be-8acb-8659e2a2925e.png">

**Comment by aamare1001** [comment]
Works on localhost, but only on American destinations, when i use an international location the api returns 'No City found'

**Comment by Nin3s** [comment]
> Works on localhost, but only on American destinations, when i use an international location the api returns 'No City found'

Is it possibly an issue with the google places api? Ik locations like london work if I put it in the weather api url directly, but rn it's taking in the trip destination as the parameter to search for

**Comment by aamare1001** [comment]
You're right it does work on london, I think the issue may be in how the trip destination is written.

So far, if it's a simple city, American State, Country(in American place's case) or city, Country(in London's case) it works, but if it's international like Sydney in Australia, (https://api.openweathermap.org/data/2.5/weather?q=**Sydney NSW, Australia**&appid=8958736f2360ef82e976ec83c5d1537d&units=imperial) it returns city not found

**Comment by aamare1001** [comment]
https://api.openweathermap.org/data/2.5/weather?q=_Sydney NSW,_ Australia&appid=8958736f2360ef82e976ec83c5d1537d&units=imperial

vs

https://api.openweathermap.org/data/2.5/weather?q=_Sydney,NSW,_Australia&appid=8958736f2360ef82e976ec83c5d1537d&units=imperial

It looks like the space is what messes with the request

**Comment by aamare1001** [comment]
Going to just remove the space from the link now that it works on production

**Comment by aamare1001** [comment]
Linked branch with this issue,

Now it checks if there is an error state and produces an error message. Also for international destinations, we take out the middle of the destination(Quebec, qc, Canada -> Quebec, Canada) in the api call for the weather. there is still a niche error with this when it comes to destinations that don't separate the first two words with a common(Sydney NSW, Australia) as the API does not recognize this, and simply taking out spaces would mess with other destinations such as Altamonte Springs, Fl, USA or would simply try and search or SydneyNSW. An idea to fix this would be to go by zip code but for now the widget will show the error message that the weather information could not be loaded to avoid crashing.

**Comment by Nin3s** [comment]
> Linked branch with this issue,
> 
> 
> 
> Now it checks if there is an error state and produces an error message. Also for international destinations, we take out the middle of the destination(Quebec, qc, Canada -> Quebec, Canada) in the api call for the weather. there is still a niche error with this when it comes to destinations that don't separate the first two words with a common(Sydney NSW, Australia) as the API does not recognize this, and simply taking out spaces would mess with other destinations such as Altamonte Springs, Fl, USA or would simply try and search or SydneyNSW. An idea to fix this would be to go by zip code but for now the widget will show the error message that the weather information could not be loaded to avoid crashing.

there is a separate api link that lets us use the zipcode of the city, but we will have to convert the city into that which I'm not really sure I know how to do atm
